src/util/rankThoughtsSequential.ts

```diff
@@ -1,3 +1,5 @@
+import { Context, Child } from "../types";
+
 /** Ranks the thoughts from 0 to n in the given array order. */
-export const rankThoughtsSequential = thoughts =>
+export const rankThoughtsSequential = (thoughts: Context) =>
```

Comment:
Technically this should be `string[]` since this operates on thoughts within a context rather than along a path.